dtformats/indx_directory_entry.py

```diff
@@ -0,0 +1,144 @@
+# -*- coding: utf-8 -*-
+"""INDX entries """
+
+import os
+
+from dtformats import data_format
+from dtformats.errors import ParseError
+
+class INDXRecord(data_format.BinaryDataFile):
+  """ Class that represents an INDX record.
+
+  Note: currently only supports $I30 index records
+  """
+
+  _FABRIC = data_format.BinaryDataFile.ReadDefinitionFile(
+    'indx_directory_entry.yml')
+
+  _DEBUG_INDX_ENTRY_HEADER = [
+    ('signature', 'signature', ''),
+    ('fixup_value_offset', 'fixup_value_offset', '_FormatIntegerAsDecimal'),
+    ('num_fixup_values', 'num_fixup_values', '_FormatIntegerAsDecimal'),
+    ('logfile_sequence_number', 'logfile_sequence_number',
+     '_FormatIntegerAsDecimal'),
+    ('virtual_cluster_number', 'virtual_cluster_number',
+     '_FormatIntegerAsDecimal')]
+
+  _DEBUG_INDX_NODE_HEADER = [
+    ('index_values_offset', 'index_values_offset',
+     '_FormatIntegerAsDecimal'),
+    ('index_node_size', 'index_node_size', '_FormatIntegerAsDecimal'),
+    ('allocated_index_node_size', 'allocated_index_node_size',
+     '_FormatIntegerAsDecimal'),
+    ('index_node_flags', 'index_node_flags', '_FormatIntegerAsDecimal')]
+
+  _DEBUG_INDX_DIR_RECORD = [
+      ('file_reference', 'file_reference', '_FormatIntegerAsDecimal'),
+      ('index_value_size', 'index_value_size', '_FormatIntegerAsDecimal'),
+      ('index_key_data_size', 'index_key_data_size', '_FormatIntegerAsDecimal'),
+      ('index_value_flags', 'index_value_flags', '_FormatIntegerAsDecimal')]
+
+  _DEBUG_FILE_NAME_ATTR = [
+      ('parent_file_reference', 'parent_file_reference',
+       '_FormatIntegerAsDecimal'),
+      ('creation_time', 'creation_time', '_FormatIntegerAsDecimal'),
+      ('modification_time', 'modification_time', '_FormatIntegerAsDecimal'),
+      ('entry_modification_time', 'entry_modification_time',
+       '_FormatIntegerAsDecimal'),
+      ('access_time', 'access_time', '_FormatIntegerAsDecimal'),
+      ('allocated_file_size', 'allocated_file_size', '_FormatIntegerAsDecimal'),
+      ('file_size', 'file_size', '_FormatIntegerAsDecimal'),
+      ('file_attribute_flags', 'file_attribute_flags',
+       '_FormatIntegerAsDecimal'),
+      ('extended_data', 'extended_data', '_FormatIntegerAsDecimal'),
+      ('name_size', 'name_size', '_FormatIntegerAsDecimal'),
+      ('name_space', 'name_space', '_FormatIntegerAsDecimal'),
+      ('filename', 'filename', '_FormatString')]
+
+  def PrintRecord(self, record):
+    """
+    Prints a human readable version of the INDX record
+    to STDOUT.
+
+    Args:
+        record (index_dir_entry): An index_dir_entry structure.
+    """
+    if record is not None:
```

Comment:
why not `if not record` ?

Reply:
I will change this to `if record` -- I am checking if `record` has a value other than `None`